projects/extension/sql/idempotent/013-vectorizer-api.sql

```diff
@@ -523,11 +526,19 @@ begin
     if _queue_schema is null or _queue_table is null then
         raise exception 'vectorizer has no queue table';
     end if;
-    select format
-    ( $sql$select count(*) from %I.%I$sql$
-    , _queue_schema, _queue_table
-    ) into strict _sql
-    ;
+    if exact_count then
+        select format
+        ( $sql$select count(1) from %I.%I$sql$
+        , _queue_schema, _queue_table
+        ) into strict _sql
+        ;
+    else
+        select format
+        ( $sql$select count(*) from (select 1 from %I.%I limit 1000)$sql$
```

Comment:
are we sure we want to return 1000 here and not say INT_MAX if the count is 1000? (btw I'd pbbl change it to 10000)

Reply:
we could also change the return type to float and return +INF

Reply:
we can also guestimate based on the relation size

Reply:
yep. or use -1 for "a lot"

Reply:
or `reltuples`